### PR TITLE
refactor test category in multiple sub groups

### PR DESCRIPTION
**Pull request by Moshe-Immerman**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -86,17 +86,15 @@ A curated list of awesome Java frameworks, libraries and software.
...
 * [Java Modeling Language (JML)](http://www.eecs.ucf.edu/~leavens/JML/) - Behavioral interface specification language useful for specifying the behavior of code modules. It combines Eiffel's Design by Contract approach with the model-based specification approach of the Larch family of interface specification languages, with some elements of the refinement calculus. Used by several other verification tools.
 * [JavaCC](https://javacc.org/) - Parser generator with syntactic lookahead.
 * [JavaServer Faces](https://javaserverfaces.java.net/) - Oracle's open-source implementation of the JSF standard, Mojarra.
 * [JavaServer Pages](https://jsp.java.net/) - Common templating for websites with custom tag libraries.
-* [JUnit](http://junit.org/) - Common testing framework.
 * [Launch4j](http://launch4j.sourceforge.net/) - Wraps JARs in lightweight and native Windows executables.
 * [Quartz](https://github.com/quartz-scheduler/quartz) - Open-source job scheduler library with Apache 2.0 license.
-* [TestNG](http://testng.org/) - Testing framework inspired by JUnit and NUnit, with different functionalities.
```
Comment (jbduncan) [inline README.md]: I agree with @heldev here. Once something has been declared "ancient", I think there needs to be a good reason to no longer declare it as such.

Consider putting JUnit and TestNG back in **Ancients**, renaming _JUnit_ to _JUnit 4_, and replacing the empty space left by removing "JUnit" from  **Testing Frameworks** with "JUnit 5" (which at this point deserves to be called _awesome_, methinks :wink:) and an appropriate description.

Comment (Moshe-Immerman) [inline README.md]: I think in this particular case, readability and usability of the list outweigh the ancient factor.  

i.e. If I am looking to take my testing to a new level, I would look for something in the same category as what I am already using (JUnit/TestNG). 

There is also an argument to be made to get rid of the ancients section entirely, by removing some of the items (e.g. JSF, JSP, Hudson, Glassfish, ant) and moving the rest to their own categories with perhaps a new "ancient" icon

Comment (heldev) [inline README.md]: Exactly, ancients section is not for "to take my testing to a new level", it's just for historic records. People are not supposed to use libraries from it.

Comment (jbduncan) [inline README.md]: Moving the entries in **Ancients** back into their respective non-ancient sections, but labelling them as "ancient" in some way, sounds compelling to me. I made a similar argument some time ago in https://github.com/akullpp/awesome-java/issues/410#issuecomment-244346612, so I wouldn't mind awesome-java going this way. However I feel @akullpp should have a say on this first before taking it further.

**Comment by akullpp** [comment]
I reworked the whole PR and decided to provide more grained categories.

Also I didn't include the new `###`-categories in the table of content since the `id` generated by GitHub would require ugly names like `Testing Miscellaneous`, `Testing Frameworks` and so on.

**Pull request merged**